### PR TITLE
feat(editor3): Add support for paste on suggesting mode (SDFID-279)

### PR DESCRIPTION
**Pull request by ioanpocol**

### CODE REVIEW

File: scripts/core/editor3/reducers/suggestions.jsx [hunk 1]
```diff
@@ -77,24 +71,77 @@ const createDeleteSuggestion = (state, {data}) => {
...
+    let {editorState} = state;
+    const blockMap = content.getBlockMap();
+
+    editorState = deleteCurrentSelection(editorState, data);
+
+    blockMap.forEach((block) => {
+        if (block.getType() !== 'atomic') {
+            const text = block.getText();
+
+            for (let i = 0; i < text.length; i++) {
```
Comment (petrjasek) [inline scripts/core/editor3/reducers/suggestions.jsx]: I think you could use here `block.characterList.forEach` and then you can check the styles using [CharacterMetadata.hasStyle](https://draftjs.org/docs/api-reference-character-metadata.html#hasstyle)

File: scripts/core/editor3/reducers/suggestions.jsx [hunk 2]
```diff
@@ -77,24 +71,77 @@ const createDeleteSuggestion = (state, {data}) => {
...
+ * @description Add a new suggestion of type ADD based on content.
+ */
+const pasteAddSuggestion = (state, {content, data}) => {
+    const acceptedStyles = ['BOLD', 'ITALIC', 'UNDERLINE'];
+    let {editorState} = state;
+    const blockMap = content.getBlockMap();
+
+    editorState = deleteCurrentSelection(editorState, data);
+
+    blockMap.forEach((block) => {
```
Comment (petrjasek) [inline scripts/core/editor3/reducers/suggestions.jsx]: I don't understand how this works - it seems like it removes the selected characters - ok - but then it set add suggestion for all non atomic blocks and every character there?

Comment (ioanpocol) [reply scripts/core/editor3/reducers/suggestions.jsx]: Yes, the suggestion entity key is set for every character.

Comment (petrjasek) [reply scripts/core/editor3/reducers/suggestions.jsx]: but how is this limited to only content which was pasted?

Comment (ioanpocol) [reply scripts/core/editor3/reducers/suggestions.jsx]: Html is converted in an editor content structure that is received here(content parameter).

Comment (petrjasek) [reply scripts/core/editor3/reducers/suggestions.jsx]: so content is a contentState with pasted content only?

Comment (ioanpocol) [reply scripts/core/editor3/reducers/suggestions.jsx]: yes

**Pull request merged**